### PR TITLE
Setup self hosted server as CI runner

### PR DESCRIPTION
**Pull request by h4n0**

### CODE REVIEW

File: ts-tests/scripts/smoke-test.ts [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  console.log(
+    `Initiating the API (ignore message "Unable to resolve type B..." and "Unknown types found...")`
+  );
+
+  // Initiate the polkadot API.
+  const api = await ApiPromise.create({
+    provider: wsProvider,
+    types: {
+      // mapping the actual specified address format
+      Address: "AccountId",
```
Comment (chenzongxiong) [inline ts-tests/scripts/smoke-test.ts:48]: what's the difference between "AccountId" and "MultiAddress"

I use "MultiAddress" in registrar. I tried "AccoundId", but failed.

File: ts-tests/scripts/smoke-test.ts [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    encodedPrefix.length + alice.addressRaw.length + encodedExpiredBlock.length
+  );
+  encodedMsg.set(encodedPrefix);
+  encodedMsg.set(alice.addressRaw, encodedPrefix.length);
+  encodedMsg.set(
+    encodedExpiredBlock,
+    encodedPrefix.length + alice.addressRaw.length
+  );
+
+  // To use manual hash and sign method, a prefix of \x19Ethereum ... is also needed to be prefixed
```
Comment (chenzongxiong) [inline ts-tests/scripts/smoke-test.ts:108]: Those comments should be removed before merged

File: ts-tests/tests/utils.ts [hunk 3]
```diff
@@ -0,0 +1,353 @@
+import 'mocha';
+
```
Comment (chenzongxiong) [inline ts-tests/tests/utils.ts:2]: Please use prettier to prettify your source code.
https://prettier.io/

Some configuration, refer to litentry-registrar or push-notification repo

**Review by chenzongxiong — CHANGES_REQUESTED** [review]

File: ts-tests/scripts/smoke-test.ts (hunk 1 above)
Comment (h4n0) [reply ts-tests/scripts/smoke-test.ts:48]: I never used MultiAddress, and this config is from the polkadot js api. Did you use the same polkadot version here?

Comment (Satoshi-Kusumoto) [reply ts-tests/scripts/smoke-test.ts:48]: Multi address is a new type of address that supports also say ethereum like address and many others.
Check https://github.com/substrate-developer-hub/substrate-node-template/pull/130

Comment (Satoshi-Kusumoto) [reply ts-tests/scripts/smoke-test.ts:48]: We also tried to introduce it in our lates parachain code #6

**Comment by h4n0** [comment]
This was done together with substrate 3.0 upgrade. Closed.

**Pull request closed without merging**